aten/src/ATen/native/transformers/cuda/sdp_utils.h

```diff
@@ -55,27 +56,31 @@ inline std::array<SDPBackend, num_backends> priority_order(sdp_params params) {
   // FlashAttention parallelizes across "batch_size * num_heads"
   // MemEff parallelizes across "batch_size * num_heads * num_queries" and can
   // be more efficient. batch_size, q_len, num_heads, k = inp.query.shape
-  if (params.query.is_nested() || params.key.is_nested() || params.value.is_nested()) {
+  if (params.query.is_nested() || params.key.is_nested() ||
+      params.value.is_nested()) {
     // See check_for_nested_inputs for details
     return {
         SDPBackend::efficient_attention,
         SDPBackend::flash_attention,
         SDPBackend::math};
   }
-  const auto sizes = params.query.sizes();
+  // const auto sizes = params.query.sizes();
   if (params.query.dim() != 4) {
     return default_order;
   }
-  const auto batch_size{sizes[0]}, num_heads{sizes[1]}, query_lengths{sizes[2]},
-      head_dim{sizes[3]};
+  const auto batch_size{params.query.sym_size(0)},
+      num_heads{params.query.sym_size(1)},
+      query_lengths{params.query.sym_size(2)},
+      head_dim{params.query.sym_size(3)};
   if (batch_size > 0) {
-    const int64_t threads_flash = batch_size * num_heads;
-    const int64_t threads_cutlass =
-        threads_flash * (int64_t)std::floor(query_lengths / 64);
```

Comment:
IIUC, this std::floor was totally unnecessary because `query_lengths` was an int and you got an int anyway.

Reply:
Yup that is my interpreation, and aftering looking at godbolt std::floor(int64_t/int64_t) and int64_t/int64_t produced the same code.